classes/submission/SubmissionDAO.inc.php

```diff
@@ -322,21 +321,28 @@ function getAssignedById($submissionId, $userId, $contextId = null, $useCache =
 			unset($submission);
 		}
 
-		$params = $this->_getFetchParameters();
-		$params[] = (int) $submissionId;
-		$params[] = (int) $userId;
+		$params = array_merge(
+			array((int) ROLE_ID_AUTHOR),
+			$this->_getFetchParameters(),
+			array((int) $submissionId)
+		);
 		if ($contextId) $params[] = (int) $contextId;
 
 		$result = $this->retrieve(
-				'SELECT	s.*, ps.date_published,
+			'SELECT	s.*, ps.date_published,
 				' . $this->_getFetchColumns() . '
-				FROM	submissions s
+			FROM	submissions s
 				LEFT JOIN published_submissions ps ON (s.submission_id = ps.submission_id)
+				LEFT JOIN stage_assignments asa ON (asa.submission_id = s.submission_id)
+				LEFT JOIN user_groups aug ON (asa.user_group_id = aug.user_group_id AND aug.role_id = ?)
 				' . $this->_getFetchJoins() . '
-				WHERE	s.submission_id = ?
-				AND s.date_submitted IS NOT NULL AND ps.date_published IS NULL AND s.user_id <> ? AND s.status <> ' .  STATUS_DECLINED .
```

Comment:
Unless I got the queries wrong, it should now appear in My Submissions if I'm assigned on the author stage, regardless if I'm the one who submitted it or not. This query gets the Assigned Submissions list, which should exclude stuff I'm assigned to as an author, because we don't want submissions to appear in both grids.